File: Android/app/src/main/java/com/example/android/NativeTurboModule.kt
package com.example.android

import android.os.Build
import com.facebook.react.bridge.ReactApplicationContext
import com.example.android.NativeTurboModuleSpec
import com.facebook.react.bridge.Promise

class NativeTurboModule(reactContext: ReactApplicationContext):NativeTurboModuleSpec(reactContext) {
    override fun getName() = NAME

    override fun getDeviceModel(promise: Promise) {
        val manufacturer: String = Build.MANUFACTURER
        val model: String = Build.MODEL
        promise.resolve(manufacturer + model)
    }
    companion object {
        const val NAME = "NativeTurboModule"
    }
}

File: ReactNative/specs/NativeTurboModule.ts
import type {TurboModule} from 'react-native';
import {TurboModuleRegistry} from 'react-native';

export interface Spec extends TurboModule {
    getDeviceModel(): Promise<string>;
}

export default TurboModuleRegistry.getEnforcing<Spec>(
  'NativeTurboModule',
);